x/incentives/types/keys.go

```diff
@@ -43,6 +43,24 @@ var (
 	// KeyIndexSeparator defines key for merging bytes
 	KeyIndexSeparator = []byte{0x07}
 
+	// KeyCurrentEpoch defines key for storing current epoch
+	KeyCurrentEpoch = []byte{0x08}
+
+	// KeyEpochBeginBlock defines key for storing begin block of current epoch
+	KeyEpochBeginBlock = []byte{0x09}
+
+	// KeyCurrentReward defines key for storing current reward
+	KeyCurrentReward = []byte{0x10} // TODO : Confirm this key name/value
+
+	// KeyHistoricalReward defines key for storing historical reward
+	KeyHistoricalReward = []byte{0x11} // TODO : Confirm this key name/value
+
+	// KeyHistoricalReward defines key for storing historical reward
+	KeyHistoricalRewardRef = []byte{0x12} // TODO : Confirm this key name/value
```

Comment:
Is explanation enough? or do you prefer to rename the key?

Reply:
I think the key name should be fine!

Reply:
It's just that I'm still confused how KeyHistoricalReward and KeyHistoricalRewardsRef are different / how they are related, connected to each other

Reply:
HistoricalReward can be found with denom, duration, and "period" which is a range of time that total stakes have not changed
But when a lock is "unlocking" or gauge becomes "finished" state, period can not be referred because
- "unlocking lock" should not be rewarded (or excluded from recent periods) if remaining locked duration is smaller than gauge's duration
- f1 distribute() will not be called if no gauge for denom&duration exist.
KeyHistoricalRewardsRef is used to get exact "period" by epoch number